src/App.js

```diff
@@ -1,51 +1,38 @@
-import React, { useState } from "react";
+import React from "react";
 import { ThemeProvider } from "styled-components";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
 import { Navbar } from "components";
 import { CoinList, CoinPage, Portfolio } from "pages";
-import { getCurrencySymbol, useLocalState } from "utils";
 import { GlobalStyle, Container, darkTheme, lightTheme } from "styling";
+import { useSelector } from "react-redux";
 import "./App.css";
 
 function App() {
-  const [currency, setCurrency] = useLocalState("currency", "usd");
-  const [isBlacked, setIsBlacked] = useLocalState("isBlacked", true);
-
-  const handleCurrencyChange = (newCurrency) => {
-    setCurrency(newCurrency);
-  };
-
-  const handleThemeChange = () => {
-    setIsBlacked(!isBlacked);
-  };
-
+  const isBlacked = useSelector( (state) => state.universalVariables.isBlacked )
+  
   return (
     <ThemeProvider theme={isBlacked ? darkTheme : lightTheme}>
       <Container>
         <GlobalStyle />
         <Router>
-          <Navbar
-            handleThemeChange={handleThemeChange}
-            handleCurrencyChange={handleCurrencyChange}
-            currency={currency}
-          />
+          <Navbar />
           <Switch>
             <Route
               exact
               path="/portfolio"
               component={(props) => (
-                <Portfolio {...props} currency={currency} />
+                <Portfolio {...props} />
               )}
             />
             <Route
               exact
               path="/coinpage/:coin"
-              component={(props) => <CoinPage {...props} currency={currency} />}
+              component={(props) => <CoinPage {...props} />}
             />
             <Route
               exact
               path="/"
-              component={(props) => <CoinList {...props} currency={currency} />}
+              component={() => <CoinList />}
```

Comment:
component={CoinList}

same for the others, the previous implementation will cause mounting and unmounting of your component every time your app re renders